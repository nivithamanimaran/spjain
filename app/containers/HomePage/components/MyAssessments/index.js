import { Typography, makeStyles, Grid } from '@material-ui/core';
import React from 'react';
import Assesment from '../../../../components/CardWidgets/Assesments';
import { useTranslation } from 'react-i18next';
import KenButton from '../../../../global_components/KenButton';
import AddIcon from '../../../../assets/icons/addPlusWhite.svg';
import { Link } from 'react-router-dom';
import Routes from '../../../../utils/routes.json';
import ProtectedComponent from '../../../../utils/rbac/ProtectedComponent';

const useStyles = makeStyles(theme => ({
  title: { fontSize: 16, fontWeight: 600 },
  container: { padding: 16 },
  link: { textDecoration: 'none' },
}));

export default function MyAssessments(props) {
  const { details, loading } = props;
  const styles = useStyles();
  const { t } = useTranslation();
  return (
    <div className={styles.container} data-testid="my-assessments" style={{padding: '0px'}}>
      <Grid container direction="row" justify="space-between">
        {/* <Grid item>
          <Typography
            className={styles.title}
            data-testid="my-assessments-label"
          >
            {t('headings:My_Assessments').toUpperCase()}&nbsp;(
            {details ? details.length : '0'})
          </Typography>
        </Grid> */}
        {/* <ProtectedComponent feature="assessments" action="add">
          <Grid item>
            <Link to={`/${Routes.assessment}`} className={styles.link}>
              <KenButton variant="primary" icon={AddIcon}>
                {t('labels:Add_New')}
              </KenButton>
            </Link>
          </Grid>
        </ProtectedComponent> */}
      </Grid>

      <Assesment details={details} loading={loading} />
    </div>
  );
}
