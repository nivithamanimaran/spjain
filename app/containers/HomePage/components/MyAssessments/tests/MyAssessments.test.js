import React, { useEffect } from 'react';
import { render, cleanup, waitForElement } from 'react-testing-library';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';
import MyAssessments from '..';
import {
  AppContextProvider,
  useAppContext,
} from '../../../../../utils/contextProvider/AppContext';

afterEach(() => {
  cleanup();
});
const tempAssessments = [
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
  {
    name: 'Assessment 1',
    subject: 'Subject 1',
    duedate: '12/11/21',
    marks: 6,
    total: 10,
    type: 'Appeared',
    url: '',
  },
];

const TestComp = () => {
  const { state, dispatch } = useAppContext();
  useEffect(() => {
    dispatch({
      type: 'updateUserDetails',
      value: {
        ContactId: '0031y00000F1W3PAAV',
        Name: 'Sanchit Faculty',
        Type: 'Faculty',
        Phone: '9405609450',
        mail: 'sanchith.faculty@ken42.com',
        address: '',
        AccountId: '0011y000008iLmlAAE',
        AccountName: 'National Center For Excellence (CVR Branch)',
        Roles: [
          {
            Id: '0031y00000F1W3PAAV',
            ClassId: 'a0F1y000000fARUEA2',
            ClassName: '2020-NCECVR-PS-ClassIII',
            AccountId: '0011y000008iLnAAAU',
            AccountName: 'NCECVR-Class III',
            department: 'NCECVR-Primary School',
            departmentId: '0011y000008iLmmAAE',
            Section: 'B',
            Role: 'Class Teacher',
          },
        ],
      },
    });
  }, []);
  return (
    <ThemeProvider>
      <Router>
        <MyAssessments details={tempAssessments} />
      </Router>
    </ThemeProvider>
  );
};

describe('My Assessment', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <MyAssessments details={tempAssessments} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <MyAssessments details={tempAssessments} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('my-assessments')).toMatchSnapshot();
  });
  it('Assessment count sould be displyed correctly', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <MyAssessments details={tempAssessments} />
        </Router>
      </ThemeProvider>
    );
    const label = getByTestId('my-assessments-label');
    const labeltext = label.textContent;
    const count = labeltext.split('(')[1].split(')')[0];
    expect(count).toBe('6');
  });
});
