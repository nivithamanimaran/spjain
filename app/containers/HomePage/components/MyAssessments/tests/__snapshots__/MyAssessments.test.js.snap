// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Assessment should render and match the snapshot 1`] = `
<div
  class="makeStyles-container-2"
  data-testid="my-assessments"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <p
        class="MuiTypography-root makeStyles-title-1 MuiTypography-body1"
        data-testid="my-assessments-label"
      >
        HEADINGS:MY_ASSESSMENTS
        Â (
        6
        )
      </p>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-18"
    data-testid="assessments"
    style="overflow: auto; max-height: 45vh;"
  >
    <div>
      <div
        class="MuiBox-root MuiBox-root-19 makeStyles-container-4"
      >
        <div
          class="MuiBox-root MuiBox-root-20 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-21 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-22 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-23 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-24 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-25 bottomDivider"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div>
                <span
                  class="MuiTypography-root makeStyles-Name-6 MuiTypography-body1"
                >
                  Assessment 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Subject 1
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-date-7 MuiTypography-body1"
              >
                12/11/21
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="show-more"
        style="padding-top: 16px;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignCenter"
        >
          Show_More
        </p>
      </div>
    </div>
  </div>
</div>
`;
